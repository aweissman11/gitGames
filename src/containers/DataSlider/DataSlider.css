.slide-btn-left,
.slide-btn-right {
  height: 100vh;
  position: absolute;
  top: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 15vw;
  transition: ease-in;
  transition-duration: 5s;
  background: white;
}

.slide-btn-left {
  left: 0;
}

.slide-btn-right {
  right: 0;
}

.slide-btn-left:hover {
  background: rgb(182,182,182);
  background: linear-gradient(90deg, rgba(182,182,182,1) 0%, rgba(255,255,255,1) 100%);
}

.slide-btn-right:hover {
  background: rgb(182,182,182);
  background: linear-gradient(90deg, rgba(255,255,255,1) 0%, rgba(182,182,182,1) 100%);
}

