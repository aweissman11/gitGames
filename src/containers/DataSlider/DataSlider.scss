@import '../../App.scss';

.data-slider {
  position: relative;
  width: 100vw;
  min-height: 100vh;
  padding-top: 70px;
  background: $my-light-black;
  z-index: 1;
}

.data-slider h3 {
  color: $my-white;
  padding: 20px;
  margin: auto;
}

// .slider-graph {

// }

.slider-graph-1 {
  height: 80vh;
  width: 80vw;
  margin: auto;
}

.slider-graph div {
  outline: none;
}

.slick-arrow {
  display: none !important;
}

.slider-btns-box {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  min-width: 100vw;
  min-height: 100vh;
}

.lft-sld-btn-box,
.rgt-sld-btn-box {
  position: sticky;
  top: 15%;
  height: 70vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  cursor: pointer;
}

.lft-sld-btn-box {
  float: left;
  padding-left: 30px;
  z-index: 100;
}

.rgt-sld-btn-box {
  float: right;
  padding-right: 30px;
  z-index: 100;
}

.left-slide-btn,
.right-slide-btn {
  height: 20px;
  width: 60px;
}

.left-slide-btn {
  z-index: 100;
  fill: $my-white;
  // left: 0;
}

.right-slide-btn {
  z-index: 100;
  fill: $my-white;
  // right: 0;
}

.right-slide-btn polygon, .slide-btn path,
.left-slide-btn polygon, .slide-btn path {
  transition: all 0.5s cubic-bezier(0.2, 1, 0.3, 1);
  fill: $my-white;
}
.rgt-sld-btn-box:hover polygon, .rgt-sld-btn-box:hover path {
  fill: rgb(220, 220, 220);
}
.lft-sld-btn-box:hover polygon, .lft-sld-btn-box:hover path {
  fill: rgb(220, 220, 220);
}

.rgt-sld-btn-box:hover polygon, .rgt-sld-btn-box:hover path,
.lft-sld-btn-box:hover polygon, .lft-sld-btn-box:hover path {
  transition: all 1s cubic-bezier(0.2, 1, 0.3, 1);
}
.rgt-sld-btn-box:hover .arrow,
.lft-sld-btn-box:hover .arrow {
  animation: arrow-anim 2.5s cubic-bezier(0.2, 1, 0.3, 1) infinite;
}
.rgt-sld-btn-box:hover .arrow-fixed,
.lft-sld-btn-box:hover .arrow-fixed {
  animation: arrow-fixed-anim 2.5s cubic-bezier(0.2, 1, 0.3, 1) infinite;
}

.slider-btns-hidden svg {
  display: none;
}

.slick-dots {
  position: static;
}

.weeks-label {
  color: $my-white;
  font-weight: bold;
  position: relative;
  bottom: 20px;
}

@keyframes arrow-anim {
  0% {
    opacity: 1;
    transform: translateX(0);
  }
  5% {
    transform: translateX(-0.1rem);
  }
  100% {
    transform: translateX(1rem);
    opacity: 0;
  }
}
@keyframes arrow-fixed-anim {
  5% {
    opacity: 0;
  }
  20% {
    opacity: 0.4;
  }
  100% {
    opacity: 1;
  }
}

