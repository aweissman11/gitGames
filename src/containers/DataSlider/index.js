import React, { Component } from 'react';
import Slider from "react-slick";

// Still might use these
// import { Route, withRouter, Switch } from 'react-router-dom';

import ForceBubbles from '../../components/ForceBubbles';
import CommitsWordCloud from '../CommitsWordCloud';
import CommitsBubble from '../CommitsBubble';
import LanguageBarChart from '../LanguageBarChart';

import { ReactComponent as LeftSlideBtn } from '../../components/LeftSlideBtn/LeftSlideBtn.svg';
import { ReactComponent as RightSlideBtn } from '../../components/RightSlideBtn/RightSlideBtn.svg';

import './DataSlider.css';

class DataSlider extends Component {

  render() {
    var settings = {
      dots: true,
      infinite: true,
      speed: 500,
      slidesToShow: 1,
      slidesToScroll: 1,
      slide: 'article',
      nextArrow: <RightSlideBtn />,
      prevArrow: <LeftSlideBtn />
    };

    return (
      <div className='data-slider'>
        <Slider {...settings} >
          <article className='slider-graph'>
            <ForceBubbles customHeight={600} />
          </article>
          <article className='slider-graph'>
            <CommitsWordCloud />
          </article>
          <article className='slider-graph'>
            <LanguageBarChart />
          </article>
          <article className='slider-graph'>
            <CommitsBubble />
          </article>
          <article className='slider-graph'>
            <h1>slider-GRAPH 5</h1>
          </article>
        </Slider>
      </div>
    );
  }
}

export default DataSlider;
