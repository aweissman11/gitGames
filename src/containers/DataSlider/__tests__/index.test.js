import React from 'react';
import { shallow } from 'enzyme';

import DataSlider from '../index';

describe('DataSlider', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<DataSlider />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
})
