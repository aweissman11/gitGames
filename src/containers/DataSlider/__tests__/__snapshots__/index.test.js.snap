// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSlider should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DataSlider />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Slider
        dots={true}
        infinite={true}
        nextArrow={<ReactComponent />}
        prevArrow={<ReactComponent />}
        slide="article"
        slidesToScroll={1}
        slidesToShow={1}
        speed={500}
      >
        <article
          className="slider-graph"
        >
          <ForceBubbles
            customHeight={768}
            customWidth={1024}
          />
        </article>
        <article
          className="slider-graph"
        >
          <h3>
            Commit Message Word Cloud
          </h3>
          <Connect(CommitsWordCloud) />
        </article>
        <article
          className="slider-graph"
        >
          <h3>
            Language Usage Breakdown
          </h3>
          <Connect(LanguageBarChart) />
        </article>
        <article
          className="slider-graph"
        >
          <h3>
            Commits broken down per week and per weekday
          </h3>
          <Connect(CommitsBarChart) />
        </article>
      </Slider>,
      "className": "data-slider",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <article
            className="slider-graph"
          >
            <ForceBubbles
              customHeight={768}
              customWidth={1024}
            />
          </article>,
          <article
            className="slider-graph"
          >
            <h3>
              Commit Message Word Cloud
            </h3>
            <Connect(CommitsWordCloud) />
          </article>,
          <article
            className="slider-graph"
          >
            <h3>
              Language Usage Breakdown
            </h3>
            <Connect(LanguageBarChart) />
          </article>,
          <article
            className="slider-graph"
          >
            <h3>
              Commits broken down per week and per weekday
            </h3>
            <Connect(CommitsBarChart) />
          </article>,
        ],
        "dots": true,
        "infinite": true,
        "nextArrow": <ReactComponent />,
        "prevArrow": <ReactComponent />,
        "slide": "article",
        "slidesToScroll": 1,
        "slidesToShow": 1,
        "speed": 500,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ForceBubbles
              customHeight={768}
              customWidth={1024}
            />,
            "className": "slider-graph",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "customHeight": 768,
              "customWidth": 1024,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Commit Message Word Cloud
              </h3>,
              <Connect(CommitsWordCloud) />,
            ],
            "className": "slider-graph",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Commit Message Word Cloud",
              },
              "ref": null,
              "rendered": "Commit Message Word Cloud",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Language Usage Breakdown
              </h3>,
              <Connect(LanguageBarChart) />,
            ],
            "className": "slider-graph",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Language Usage Breakdown",
              },
              "ref": null,
              "rendered": "Language Usage Breakdown",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Commits broken down per week and per weekday
              </h3>,
              <Connect(CommitsBarChart) />,
            ],
            "className": "slider-graph",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Commits broken down per week and per weekday",
              },
              "ref": null,
              "rendered": "Commits broken down per week and per weekday",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "article",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Slider
          dots={true}
          infinite={true}
          nextArrow={<ReactComponent />}
          prevArrow={<ReactComponent />}
          slide="article"
          slidesToScroll={1}
          slidesToShow={1}
          speed={500}
        >
          <article
            className="slider-graph"
          >
            <ForceBubbles
              customHeight={768}
              customWidth={1024}
            />
          </article>
          <article
            className="slider-graph"
          >
            <h3>
              Commit Message Word Cloud
            </h3>
            <Connect(CommitsWordCloud) />
          </article>
          <article
            className="slider-graph"
          >
            <h3>
              Language Usage Breakdown
            </h3>
            <Connect(LanguageBarChart) />
          </article>
          <article
            className="slider-graph"
          >
            <h3>
              Commits broken down per week and per weekday
            </h3>
            <Connect(CommitsBarChart) />
          </article>
        </Slider>,
        "className": "data-slider",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <article
              className="slider-graph"
            >
              <ForceBubbles
                customHeight={768}
                customWidth={1024}
              />
            </article>,
            <article
              className="slider-graph"
            >
              <h3>
                Commit Message Word Cloud
              </h3>
              <Connect(CommitsWordCloud) />
            </article>,
            <article
              className="slider-graph"
            >
              <h3>
                Language Usage Breakdown
              </h3>
              <Connect(LanguageBarChart) />
            </article>,
            <article
              className="slider-graph"
            >
              <h3>
                Commits broken down per week and per weekday
              </h3>
              <Connect(CommitsBarChart) />
            </article>,
          ],
          "dots": true,
          "infinite": true,
          "nextArrow": <ReactComponent />,
          "prevArrow": <ReactComponent />,
          "slide": "article",
          "slidesToScroll": 1,
          "slidesToShow": 1,
          "speed": 500,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ForceBubbles
                customHeight={768}
                customWidth={1024}
              />,
              "className": "slider-graph",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "customHeight": 768,
                "customWidth": 1024,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Commit Message Word Cloud
                </h3>,
                <Connect(CommitsWordCloud) />,
              ],
              "className": "slider-graph",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Commit Message Word Cloud",
                },
                "ref": null,
                "rendered": "Commit Message Word Cloud",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Language Usage Breakdown
                </h3>,
                <Connect(LanguageBarChart) />,
              ],
              "className": "slider-graph",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Language Usage Breakdown",
                },
                "ref": null,
                "rendered": "Language Usage Breakdown",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Commits broken down per week and per weekday
                </h3>,
                <Connect(CommitsBarChart) />,
              ],
              "className": "slider-graph",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Commits broken down per week and per weekday",
                },
                "ref": null,
                "rendered": "Commits broken down per week and per weekday",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "article",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
